### PR TITLE
Refactor to improve types for `max-*` rules

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Part of #4496

> Is there anything in the PR that needs further explanation?

This change removes `// @ts-nocheck` from the `max-*` rules.

### CODE REVIEW

**Comment by ybiquitous** [comment]
### To-do

- [ ] Wait for merging #5507

File: lib/rules/max-empty-lines/index.js [hunk 1]
```diff
@@ -67,18 +64,19 @@ function rule(max, options, context) {
 			const rootRawsAfter = root.raws.after;
 
 			// not document node
+			// @ts-expect-error -- TS2339: Property 'document' does not exist on type 'Root'.
```
Comment (jeddy3) [inline lib/rules/max-empty-lines/index.js:67]: We'll likely need to revisit this `document` stuff at some point in light of https://github.com/postcss/postcss/pull/1582.

I think it's mainly the max whitespace rules that touch it.

**Review by jeddy3 — APPROVED** [review]
Looks great, thanks!

File: lib/rules/max-empty-lines/index.js (hunk 1 above)
Comment (ybiquitous) [reply lib/rules/max-empty-lines/index.js:67]: Yes, I also think so. 👍🏼 
I didn't know how I should handle it, so I've left `// @ts-expect-error` comments.

Comment (jeddy3) [reply lib/rules/max-empty-lines/index.js:67]: Leaving `// @ts-expect-error` comments was the right thing to do.

Feel free to merge.

Comment (ybiquitous) [reply lib/rules/max-empty-lines/index.js:67]: Thank you!

**Pull request merged**